test/SerialIOTest.cpp

```diff
@@ -899,6 +899,26 @@ TEST_CASE( "hdf5_dtype_test", "[serial][hdf5]" )
         s.setAttribute("vecLongdouble", std::vector< long double >({0.L, std::numeric_limits<long double>::max()}));
         s.setAttribute("vecString", std::vector< std::string >({"vector", "of", "strings"}));
         s.setAttribute("bool", true);
+
+        // translation from non-fixed size integer types
+        int rs = 8;
+        s.setAttribute("short", rs);
+        int i = 16;
+        s.setAttribute("int", i);
+        long l = 32;
+        s.setAttribute("long", l);
+        unsigned short us = 8u;
+        s.setAttribute("ushort", us);
+        unsigned int ui = 16u;
+        s.setAttribute("uint", ui);
+        unsigned long ul = 32u;
+        s.setAttribute("ulong", ul);
+        s.setAttribute("vecShort", std::vector< short >({32766, 32767}));
+        s.setAttribute("vecInt", std::vector< int >({32766, 32767}));
+        s.setAttribute("vecLong", std::vector< long >({2147483646, 2147483647}));
+        s.setAttribute("vecUShort", std::vector< unsigned short >({65534u, 65535u}));
+        s.setAttribute("vecUInt", std::vector< unsigned int >({65533u, 65531u}));
+        s.setAttribute("vecULong", std::vector< unsigned long >({65532u, 65530u}));
```

Comment:
There already is a testcase in [```CoreTest.cpp```](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/test/CoreTest.cpp#L26-L80) that does similar checks. As this is (probably?) not supposed to check the workings in the backend, and rather in ```Attribute```, the tests are better suited there.

Reply:
Ah dang, no... I moved the test to `CoreTest` now where a (non-user typical) `Attribute()` is created directly. I did not fix that yet ([see reason below](https://github.com/openPMD/openPMD-api/pull/333#issuecomment-419088770)). I fixed the user-interface `setAttribute`.